src/tar.ts

```diff
@@ -49,13 +32,27 @@ function getWorkingDirectory(): string {
     return process.env["GITHUB_WORKSPACE"] ?? process.cwd();
 }
 
-export async function extractTar(archivePath: string): Promise<void> {
+function isOS64(): boolean {
+    return process.platform != "win32" || process.arch === "x64";
+}
+
+export async function extractTar(
+    archivePath: string,
+    compressionMethod: CompressionMethod
+): Promise<void> {
     // Create directory to extract tar into
     const workingDirectory = getWorkingDirectory();
     await io.mkdirP(workingDirectory);
+    // --d: Decompress.
+    // --long=#: Enables long distance matching with # bits. Maximum is 30 (1GB) on 32-bit OS and 31 (2GB) on 64-bit.
     const args = [
-        "-xz",
-        "-f",
+        ...(compressionMethod == CompressionMethod.Zstd
+            ? [
+                  "--use-compress-program",
+                  isOS64() ? "zstd -d --long=31" : "zstd -d --long=30"
```

Comment:
Does this have any impact on compatibility between 32bit and 64bit OS's using the same cache?

I.e. Can a cache created on a 64-bit runner be extracted by a 32-bit runner?

If so, we may want to include the OS arch in the cache version hash

Reply:
You are right it would. 

I think I'm gonna change it to use the universal 30. It wouldn't be the fastest but having too many version of essentially the same cache is also not ideal.

Reply:
Are you going to make any changes related to long distance matching in this PR?

Reply:
Yes. I'm changing it to 30 in the next commit unless someone strongly oppose (and with good reasoning of course)

Reply:
Okay, I think that's fine.

Reply:
Another option is to drop down to 30 across the board.

This would eliminate code complexity (by dropping the architecture bit-size checks and branches), and would avoid wasting storage caused by cache users exercising mixed-bit-size platforms.

The trade off is giving up the the compression benefits of finding matching across 2GB spans of data versus 1GB spans; which for some highly-redundant data (say, like ) can make a big difference, but for binary / code / etc.. it's not going to be a big loss.



-  forfeit a very marginal amount of compression